### PR TITLE
Gherkin: customization of available roles in project membership 

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-4650
Story: https://issues.redhat.com/browse/ODC-5438

Acceptance criteria:

- As an administrator, I want to customize the roles which are being shown in the Project Access tab in the Developer console.  This includes removing default roles as well as adding custom roles. 

**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [x] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        Scenario: Adding Custom role in project membership: A-09-TC01
+            Given user has cluster role basic-user created
+             When user clicks on Search tab in navigation menu
+              And user searches console in Resources dropdown
+              And user selects Console with apiVersion operator.openshift.io/v1 option
+              And user clicks on "cluster" Name in Consoles
+              And user switches to YAML tab
+              And user adds spec.customization.projectAccess section
+              And user inserts Add project access roles Snippet from the Sidebar under spec.customization.projectAccess section
+              And user adds - basic-user under the other three role added by Snippet
```
Comment (vikram-raj) [inline frontend/packages/dev-console/integration-tests/features/project/project-access.feature:20]:
```suggestion
              And user adds "- basic-user" under the other three role added from Snippet
```

Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/project/project-access.feature:20]: Updated

**Review by vikram-raj — COMMENTED** [review]
/lgtm

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature [hunk 2]
```diff
@@ -0,0 +1,38 @@
+Feature: Project Access page
+              As a administrator, I want to customize the roles which are being shown in the Project Access tab in the Developer console, this includes removing default roles as well as adding custom roles.
+
+
+        Background:
+            Given user is at developer perspective
+              And user has created or selected namespace "aut-project-access"
+
+
+        @regression, @odc-4650, @manual
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/project/project-access.feature:10]: remove the "," between tags

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+              And user adds "- basic-user" under the other three role added from Snippet
+              And user clicks on Save button
+              And user clicks on Project tab in navigation menu
+              And user clicks Project Access tab
+              And user clicks on Role dropdown for kube:admin Name
+             Then user will see "basic-user" option in the dropdown
+
+
+
+        @regression, @odc-4650, @manual
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/project/project-access.feature:29]: remove the "," between the tags

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            Given user is at developer perspective
+              And user has created or selected namespace "aut-project-access"
+
+
+        @regression, @odc-4650, @manual
+        Scenario: Adding Custom role in project membership: A-09-TC01
+            Given user has cluster role basic-user created
+             When user clicks on Search tab in navigation menu
+              And user searches console in Resources dropdown
+              And user selects Console with apiVersion operator.openshift.io/v1 option
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/project/project-access.feature:15]:
```suggestion
              And user selects Console with apiVersion operator.openshift.io/v1 option from Resources dropdown
```

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+              And user clicks on "cluster" Name in Consoles
+              And user switches to YAML tab
+              And user adds spec.customization.projectAccess section
+              And user inserts Add project access roles Snippet from the Sidebar under spec.customization.projectAccess section
+              And user adds "- basic-user" under the other three role added from Snippet
+              And user clicks on Save button
+              And user clicks on Project tab in navigation menu
+              And user clicks Project Access tab
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/project/project-access.feature:23]:
```suggestion
              And user clicks Project access tab
```

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+              As a administrator, I want to customize the roles which are being shown in the Project Access tab in the Developer console, this includes removing default roles as well as adding custom roles.
+
+
+        Background:
+            Given user is at developer perspective
+              And user has created or selected namespace "aut-project-access"
+
+
+        @regression, @odc-4650, @manual
+        Scenario: Adding Custom role in project membership: A-09-TC01
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/project/project-access.feature:11]:
```suggestion
        Scenario: Adding Custom role in project membership: PA-01-TC01
```

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+              And user clicks on Save button
+              And user clicks on Project tab in navigation menu
+              And user clicks Project Access tab
+              And user clicks on Role dropdown for kube:admin Name
+             Then user will see "basic-user" option in the dropdown
+
+
+
+        @regression, @odc-4650, @manual
+        Scenario: Removing Custom role in project membership: A-09-TC01
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/project/project-access.feature:30]:
```suggestion
        Scenario: Removing Custom role in project membership: PA-01-TC02
```

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature [hunk 8]
```diff
@@ -0,0 +1,38 @@
+Feature: Project Access page
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/project/project-access.feature:1]: Add @project-access before this line

**Review by makambalaji — CHANGES_REQUESTED** [review]
Add this template: https://github.com/openshift/console/blob/master/frontend/packages/dev-console/integration-tests/features/.github/PULL_REQUEST_TEMPLATE/qe-pr.md  in pr description

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature (hunk 3 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/project/project-access.feature:29]: Updated

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature (hunk 4 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/project/project-access.feature:15]: Updated

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature (hunk 5 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/project/project-access.feature:23]: Updated

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature (hunk 6 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/project/project-access.feature:11]: Updated

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature (hunk 7 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/project/project-access.feature:30]: Updated

File: frontend/packages/dev-console/integration-tests/features/project/project-access.feature (hunk 8 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/project/project-access.feature:1]: Updated

**Pull request merged**